python/cudf_polars/tests/test_scan.py

```diff
@@ -400,6 +400,28 @@ def test_scan_parquet_chunked(
     )
 
 
-def test_scan_hf_url_raises():
-    q = pl.scan_csv("hf://datasets/scikit-learn/iris/Iris.csv")
-    assert_ir_translation_raises(q, NotImplementedError)
+@pytest.mark.parametrize(
+    "file_ext, writer, scan",
+    [
+        ("parquet", "write_parquet", pl.scan_parquet),
+        ("csv", "write_csv", pl.scan_csv),
+        ("ndjson", "write_ndjson", pl.scan_ndjson),
+    ],
+)
+def test_scan_include_file_paths(tmp_path, file_ext, writer, scan):
+    df1 = pl.DataFrame({"a": [1, 2, 3]})
+    df2 = pl.DataFrame({"a": [7, 6, 5]})
+
+    file1 = tmp_path / f"df1.{file_ext}"
+    file2 = tmp_path / f"df2.{file_ext}"
+
+    getattr(df1, writer)(file1)
+    getattr(df2, writer)(file2)
+
+    q = (
+        scan([file1, file2], include_file_paths="path")
+        .with_row_index()
+        .select(pl.col("a"))
+    )
+
+    assert_gpu_result_equal(q, engine=NO_CHUNK_ENGINE)
```

Comment:
Since the file paths are dropped in the `select` call (as is the row index), we haven't checked that we got the right result here, I think.

Also, does this only work for the unchunked engine?